### PR TITLE
Remove SDL_cdrom based CD-ROM interfaces

### PR DESCRIPTION
**Pull request by dreamer**
Another step towards #29.

Less code got removed then for previous interfaces, but much bigger impact. Removal of code opened opportunities for various cleanup and simplifications - some of them I implemented in this PR already, but I stopped myself, because new areas for cleanup started to reveal.

SDL_cdrom was a basis for whole physical CD-ROM support - all other interfaces used at least part of functionality from this lib.

(I suggest reviewing commits one-by-one, otherwise it's hard to follow what was moved where and why)

### CODE REVIEW

File: src/dos/dos_mscdex.cpp [hunk 1]
```diff
@@ -253,18 +253,11 @@ int CMscdex::AddDrive(Bit16u _drive, char* physicalPath, Bit8u& subUnit)
...
-		// Always use IOCTL in Linux
-		cdrom[numDrives] = new CDROM_Interface_Ioctl();
-		LOG(LOG_MISC,LOG_NORMAL)("MSCDEX: IOCTL Interface.");
-#else
-		// Default case windows and other oses
-		cdrom[numDrives] = new CDROM_Interface_SDL();
-		LOG(LOG_MISC,LOG_NORMAL)("MSCDEX: SDL Interface.");
-#endif
-		} break;
+		// TODO: support for mounting physical CD-ROMs removed, provide
```
Comment (krcroft) [inline src/dos/dos_mscdex.cpp]: I like this approach.
We can point user to a README or a Wiki page up-to-date with tools/scripts showing how to backup their game CD-ROM to ISO/FLACs & ISO/Opus tracks.

File: README [hunk 2]
```diff
@@ -469,7 +464,7 @@ To get a list of the internal commands type "HELP" at the prompt.
 In addition, the following commands are available:
 
 MOUNT "Emulated Drive letter" "Real Drive or Directory"
-      [-t type] [-noioctl] [-usecd number] [-size drivesize]
+      [-t type] [-usecd number] [-size drivesize]
```
Comment (krcroft) [inline README]: Knock out `-usecd` and `-cd` references.

File: README [hunk 3]
```diff
@@ -506,20 +501,14 @@ MOUNT -u "Emulated Drive letter"
...
           For Windows: label is extracted from "Real Drive".
           For Linux: label is set to NO_LABEL.
 
         If you do specify a label, this label will be kept as long as the drive
         is mounted. It will not be updated !!
 
-  -noioctl
-        Forces use of the SDL CD-ROM layer. Valid on all systems.
-
   -usecd number
```
Comment (krcroft) [inline README]: (ditto)

File: README [hunk 4]
```diff
@@ -506,20 +501,14 @@ MOUNT -u "Emulated Drive letter"
...
-        Valid on all systems, under Windows the -noioctl switch has to be
-        present to make use of the -usecd switch.
         Enables to select the drive that should be used by SDL. Use this if
         the wrong or no CD-ROM drive is mounted while using the SDL CD-ROM
         interface. "number" can be found by "MOUNT -cd".
```
Comment (krcroft) [inline README]: (ditto)

File: src/dos/dos_mscdex.cpp [hunk 5]
```diff
@@ -253,18 +253,11 @@ int CMscdex::AddDrive(Bit16u _drive, char* physicalPath, Bit8u& subUnit)
 	int result = 0;
 	// Get Mounttype and init needed cdrom interface
 	switch (CDROM_GetMountType(physicalPath,forceCD)) {
-	case 0x00: {	
+	case 0x00:
 		LOG(LOG_MISC,LOG_NORMAL)("MSCDEX: Mounting physical cdrom: %s"	,physicalPath);
```
Comment (krcroft) [inline src/dos/dos_mscdex.cpp]: Reword `physical cdrom` (only can mount directories and CUEs now)

File: README [hunk 6]
```diff
@@ -132,15 +132,10 @@ CD-ROM: My CD-ROM doesn't work.
     in Linux:
       - mount d /media/cdrom -t cdrom
```
Comment (krcroft) [inline README]: The distinction now is that `/media/cdrom` needs to be a filesystem as opposed to block device (and `/media/cdrom` is a bit ambiguous now because some distro's those are aliases to the devices like `/dev/sr0`).

Suggest something like: mount d /path/to/cdrom -t cdrom

That way, the user can mount their CD-ROM host-side and DOSBox can still access it "directly" (albeit without CD-DA support).

File: src/dos/dos_programs.cpp [hunk 7]
```diff
@@ -323,12 +322,15 @@ class MOUNT : public Program {
 			if (type=="cdrom") {
 				int num = -1;
 				cmd->FindInt("-usecd",num,true);
```
Comment (krcroft) [inline src/dos/dos_programs.cpp]: Knockout `-usecd`

File: src/dos/dos_programs.cpp [hunk 8]
```diff
@@ -319,19 +317,15 @@ class MOUNT : public Program {
...
-				if (cmd->FindExist("-ioctl", false)) {
-					WriteOut(MSG_Get("MSCDEX_WARNING_IOCTL"));
-				}
-				if (cmd->FindExist("-noioctl", false)) {
-					WriteOut(MSG_Get("MSCDEX_WARNING_IOCTL"));
+
+			if (type == "cdrom") {
+				// Following options were relevant only for physical CD-ROM  support:
+				for (auto opt : {"-usecd", "-noioctl", "-ioctl", "-ioctl_dx", "-ioctl_mci", "-ioctl_dio"}) {
+					if (cmd->FindExist(opt, false))
```
Comment (krcroft) [inline src/dos/dos_programs.cpp]: I'm not sure if `-cd` can be provided a similar message; altough it's a stand-alone flag given to `mount`, so wouldn't fall under this `cdrom` condition.

File: src/dos/cdrom.h [hunk 9]
```diff
@@ -120,35 +121,46 @@ class CDROM_Interface_Image : public CDROM_Interface
...
 
 	class AudioFile : public TrackFile {
 	public:
 		AudioFile       (const char *filename, bool &error);
+		~AudioFile      ();
+
+		AudioFile       () = delete;
+		AudioFile       (const AudioFile&) = delete; // prevent copying
+		AudioFile&      operator= (const AudioFile&) = delete; // prevent assignment
+
```
Comment (krcroft) [inline src/dos/cdrom.h]: :+1:

File: src/dos/cdrom.h [hunk 10]
```diff
@@ -59,19 +60,22 @@ typedef struct SCtrl {
 
 // Conversion function from frames to Minutes/Second/Frames
 //
-template<typename T>
-inline void frames_to_msf(int frames, T *m, T *s, T *f) {
-	*f = frames % REDBOOK_FRAMES_PER_SECOND;
+inline TMSF frames_to_msf(int frames)
+{
```
Comment (krcroft) [inline src/dos/cdrom.h]: Love this refactoring here; simpler and more readable.

File: src/dos/dos_mscdex.cpp (hunk 5 above)
Comment (dreamer) [reply src/dos/dos_mscdex.cpp]: This case (0x00) is still responsible for physical CD-ROMs; in later commit I introduced an enum to make it clear, but I think I can safely remove enum value and this case handling now. I will do it in a new commit.

File: README (hunk 6 above)
Comment (dreamer) [reply README]: Many distros standardized on using `/run/media/<user>/<drive_label>` now, but I agree - I'll replace it with `/path/to/cdrom`.

File: src/dos/dos_programs.cpp [hunk 11]
```diff
@@ -197,7 +197,7 @@ class MOUNT : public Program {
 		}
 
 		if (cmd->FindExist("-cd", false)) {
-   			WriteOut(MSG_Get("PROGRAM_MOUNT_NO_CDROM_SUPPORT"));
+   			WriteOut(MSG_Get("PROGRAM_MOUNT_NO_OPTION"), "-cd");
```
Comment (krcroft) [inline src/dos/dos_programs.cpp]: Much more helpful than the original (which didn't explicitly say which of possibly-multiple options, was the problem)

**Comment by dreamer** [comment]
Thanks for the review :) GitHub automatically dismissed the review after I applied fixup commits, but I don't think we need to re-review it. I'll merge it in as soon as CI will finish.
